### DISCUSSION TITLE
Ring offset default to match background color

### DISCUSSION THREAD

**Discussion by WesleyKapow** [tailwindlabs/tailwindcss#8421] [Ideas]
Seems common to want to set the `ring-offset` color to match the background.

In my code I have:
```css
.bg-gray-100 button,
.bg-gray-100 a {
  @apply focus-visible:ring-offset-gray-100
}

.bg-gray-200 button,
.bg-gray-200 a {
  @apply focus-visible:ring-offset-gray-200
}

/* ...etc for all used background colors */
```

What if instead tailwind:
1. Changed the way bg classes were computed to include a CSS variable for the set background, e.g. `--tw-background-color` 
2. Set the default ring-offset color to `--tw-background-color`.

This would allow the ring offset color to automatically pick up the parent's bg.

Thoughts? Is there a way I can extend how background classes are computed to do this in my code/config?

**Comment by simon-abbott** [comment]
I totally second this. A hard default of `#fff` is not a great experience unless all your ringed elements happen to be on a white background. It makes far more sense in my mind for the default color to be whatever the current background color is.

I put together [some rough patches](https://gist.github.com/simon-abbott/448955df5a1963fc54ae99069a13ef57) you can apply with [patch-package](https://www.npmjs.com/package/patch-package) if you want this behavior now. Obviously this is unsupported, so use with caution.
